includes/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Promoted_Jobs_API.
+ *
+ * @package wp-job-manager
+ */
+
+/**
+ * Handles functionality related to the Promoted Jobs REST API.
+ *
+ * @since $$next-version$$
+ */
+class WP_Job_Manager_Promoted_Jobs_API extends WP_REST_Controller {
+
+	/**
+	 * The namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace;
+
+	/**
+	 * Rest base for the current object.
+	 *
+	 * @var string
+	 */
+	protected $rest_base;
+
+	/**
+	 * Associated post meta.
+	 *
+	 * @var string
+	 */
+	protected $meta;
+
+	/**
+	 * Promoted Jobs Class constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'wpjm-internal/v1';
+		$this->rest_base = 'promoted-jobs';
+		$this->meta      = '_promoted';
+	}
+
+	/**
+	 * Register the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' .
+			$this->rest_base,
+			[
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_items' ],
+					'permission_callback' => '__return_true',
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get all promoted jobs.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function get_items( $request ) {
+		$args = [
+			'post_type'           => 'job_listing',
+			'post_status'         => 'publish',
+			'no_found_rows'       => true,
+			'ignore_sticky_posts' => true,
+			'meta_query'          => [
+				[
+					'key'     => $this->meta,
+					'value'   => '1',
+					'compare' => '==',
+				],
+			],
+		];
+
+		$job_listings = new WP_Query( $args );
+		$items        = $job_listings->posts;
+		$data         = [];
+
+		if ( empty( $items ) ) {
+			return rest_ensure_response( $data );
+		}
+
+		foreach ( $items as $item ) {
+			$itemdata = $this->prepare_item_for_response( $item, $request );
+			$data[]   = $this->prepare_response_for_collection( $itemdata );
+		}
+
+		return new WP_REST_Response( $data, 200 );
```

Comment:
@mikeyarce, sorry for this late suggestion here, but I noticed something while working on wpjobmanager.com pointing to this PR.

It would be nice if we add a property here to have the array, instead of returning it directly. Maybe `jobs`?

So it gets easier if we need to expand this endpoint in the future adding some extra information.

Reply:
Good idea!  Done in 86f948e7d8002352740f0fb718533af78075e1a9